### PR TITLE
Crud

### PR DESCRIPTION
**Pull request by j-tirelli**
From Legacy code, I made the following changes...

1. Moved DB functions from the server file into a helper function file.
1. Changed the logic from sending network requests to itself 3 times for each page load to source items to create a list of similiar items, to only querying the database directly once, by dept (which has so many hits that the others items were never seen anyway). 
1. Added routes and methods for CRUD operations.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -11,52 +12,101 @@ app.use(express.urlencoded({ extended: true }));
...
     res.json(results);
   });
 });
 
-app.get('/products/brand/:brandName', (req, res) => {
-  // escape certain characters from request url
-  let brandWords = req.params.brandName.split(/[,.\s-&amp]/);
-  RecommendedItem.find({
-    brand: {$regex: `^${brandWords.join('.*\s*')}$`, $options: 'i'}}, (err, results) => {
+app.get('/products/price/min=:minPrice&max=:maxPrice', async (req, res) => {
```
Comment (luke-eastman) [inline server/index.js:29]: I like your use of the params object here. Seems much cleaner and transparent than what I've been doing.

File: database/helpers.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+const RecommendedItem = require('./RecommendedItem.js');
+
+const formatName = (string) => {
+  return string[0].toUpperCase() + string.split('').slice(1).join('');
+};
+
+const getDept = (dept, callback = (result) => result) => {
+  let department = formatName(dept);
+  return RecommendedItem.find({ department }, callback);
```
Comment (luke-eastman) [inline database/helpers.js]: Is a return necessary here? Not that it is causing any problems. Just curious.

File: database/helpers.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+};
+
+const getBrands = (brandName, callback = (result) => result) => {
+  let brandWords = brandName.split(/[,.\s-&amp]/);
+  let $regex = `^${brandWords.join('.*\s*')}$`;
+  let $options = 'i'
+  return RecommendedItem.find({ brand: { $regex, $options }}, callback);
+};
+
+const getPrices = ($gte = 0, $lte = 1000, callback = (result) => result) => {
```
Comment (luke-eastman) [inline database/helpers.js:19]: Good use of defaults. Something I should be doing more of.

File: server/index.js [hunk 4]
```diff
@@ -11,52 +12,98 @@ app.use(express.urlencoded({ extended: true }));
...
-    price: {
-      $gte: req.params.minPrice || 0,
-      $lte: req.params.maxPrice || 1000 }
-    }, (err, results) => {
-      res.json(results);
+app.get('/products/id/:productId', async (req, res) => {
+  helpers.getProduct(req.params.productId, (err, searchedProduct) => {
+    if (err) {
+      console.error(err);
+      res.status(503).send('Error Caught at findOne a callback');
```
Comment (luke-eastman) [inline server/index.js:39]: I think I need to look more into different error codes.

**Review by luke-eastman — APPROVED** [review]
This all looks really good. You already fixed the things I was wondering about for the most part. If anything, I think I got more out of reviewing this pr than you did.

File: database/helpers.js (hunk 2 above)
Comment (j-tirelli) [reply database/helpers.js]: No, not at all. Good catch that was a relic of multiple trial and errors! I'll remove them.

File: server/index.js (hunk 4 above)
Comment (j-tirelli) [reply server/index.js:39]: Ha! I was just guessing here..

**Pull request merged**